sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/implementation/models/MessageIdsUpdateHeaders.java

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.storage.queue.implementation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.util.DateTimeRfc1123;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+import java.time.OffsetDateTime;
+
+/** The MessageIdsUpdateHeaders model. */
+@JacksonXmlRootElement(localName = "null")
+@Fluent
+public final class MessageIdsUpdateHeaders {
+    /*
+     * The x-ms-version property.
+     */
+    @JsonProperty(value = "x-ms-version")
+    private String xMsVersion;
+
+    /*
+     * The x-ms-time-next-visible property.
+     */
+    @JsonProperty(value = "x-ms-time-next-visible")
+    private DateTimeRfc1123 xMsTimeNextVisible;
+
+    /*
+     * The x-ms-request-id property.
+     */
+    @JsonProperty(value = "x-ms-request-id")
+    private String xMsRequestId;
+
+    /*
+     * The x-ms-popreceipt property.
+     */
+    @JsonProperty(value = "x-ms-popreceipt")
+    private String xMsPopreceipt;
+
+    /*
+     * The Date property.
+     */
+    @JsonProperty(value = "Date")
+    private DateTimeRfc1123 dateProperty;
+
+    /**
+     * Get the xMsVersion property: The x-ms-version property.
+     *
+     * @return the xMsVersion value.
+     */
+    public String getXMsVersion() {
```

Comment:
Is there a way to change these method names to not have XMs?

Reply:
@srnagar and @jianghaolu Is there a setting in the generator where we can fix that?

Reply:
For now you will have to use the customization to rename a method.

Reply:
It seems like a bug in autorest core. You have `x-ms-client-name` defined(https://github.com/Azure/azure-rest-api-specs/blob/storage-dataplane-preview/specification/storage/data-plane/Microsoft.QueueStorage/preview/2018-03-28/queue.json#L70) and the autorest core docs actually used this `xMsVersion` as an example: https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-name

Reply:
@jianghaolu Should I create an issue un autorest core?